src/utils.rs

```diff
@@ -0,0 +1,84 @@
+// All code cribbed from num-integer. We do this to minimize dependencies and
```

Comment:
When cribbing other software, something to be cautious around is declaring licenses. On advice of a lawyer, it was adequate to surround code with explicit comments about the source, and the license*.

Here is an example from my own project https://github.com/jtmoon79/super-speedy-syslog-searcher/blob/0.5.58/src/data/utmpx.rs#L134-L186

I recommend doing the same here. The top of the file should include a docstring stating something like

```lang-rust
//! Number utility functions.
//!
//! Code in this file `src/utils.rs` was copied from project [`num-integer`].
//! That project is [licensed under the MIT license].
//!
//! [`num-integer`]: https://github.com/rust-num/num-integer
//! [licensed under the MIT license]: https://github.com/rust-num/num-integer/blob/num-integer-0.1.45/LICENSE-MIT
```

_* not formal legal advice, just a recommendation_